zulipterminal/ui_tools/boxes.py

```diff
@@ -636,10 +636,34 @@ def autocomplete_streams(
 
         prefix_length = len(prefix_string)
 
-        matched_data = match_stream(
+        _, matched_streams = match_stream(
             stream_data, text[prefix_length:], self.view.pinned_streams
         )
-        return matched_data
+
+        muted_stream_names = [
+            stream_name
+            for stream_name in matched_streams
+            if stream_name
+            in [
+                self.model.stream_dict[stream_id]["name"]
+                for stream_id in self.model.muted_streams
+            ]
+        ]
+
+        for muted_stream in muted_stream_names:
+            matched_streams.remove(muted_stream)
+            if muted_stream in [stream["name"] for stream in self.view.pinned_streams]:
+                muted_stream_names.remove(muted_stream)
+                muted_stream_names.insert(0, muted_stream)
+        matched_streams.extend(muted_stream_names)
+
+        current_stream_name = self.model.stream_dict[self.stream_id]["name"]
```

Comment:
This fails for private messages, and actually causes a crash. There is an outstanding question of what should be the stream to use here - narrow? compose? recent? neither?